src/compiler/resolutionCache.ts

```diff
@@ -561,14 +561,15 @@ namespace ts {
             resolution: T,
             filePath: Path,
             getResolutionWithResolvedFileName: GetResolutionWithResolvedFileName<T, R>,
+            noResolveResolution: T
         ) {
             if (resolution.refCount) {
                 resolution.refCount++;
                 Debug.assertDefined(resolution.files);
             }
             else {
                 resolution.refCount = 1;
-                Debug.assert(resolution.files === undefined);
+                Debug.assert(resolution.files === undefined || (resolution === noResolveResolution && length(noResolveResolution.files) === 0));

```

Comment:
Do `undefined` and `[]` have different meanings for `noResolveResolution.files`?  If not, why not just initialize it to `undefined` and create it the first time something is added?  (Mostly, it seems weird to me that `resolution` and `noResolveResolution` don't have the same shape.)

Reply:
Instead of that i just changed the assert as we are really checking if no file is referencing this resolution.